app/src/modules/insights/components/dashboard-dialog.vue

```diff
@@ -29,83 +29,77 @@
 	</v-dialog>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import api from '@/api';
-import { unexpectedError } from '@/utils/unexpected-error';
-import { defineComponent, ref, reactive, PropType, watch } from 'vue';
-import { useInsightsStore } from '@/stores/insights';
 import { router } from '@/router';
+import { useInsightsStore } from '@/stores/insights';
 import { Dashboard } from '@/types/insights';
-import { useI18n } from 'vue-i18n';
+import { unexpectedError } from '@/utils/unexpected-error';
 import { isEqual } from 'lodash';
+import { reactive, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+const props = withDefaults(
+	defineProps<{
+		modelValue?: boolean;
+		dashboard?: Dashboard;
+	}>(),
+	{
+		modelValue: false,
```

Comment:
- Remove unnecessary `modelValue: false`. Booleans default to `false` automatically. Or do we wan't to be explicit here since it is a modelValue and not a simple flag?